multires-lyrics-search/tests/test_multireslyrics.py

```diff
@@ -0,0 +1,224 @@
+    os.environ.setdefault("JINA_DATA_FILE", "tests/data-index.csv")
+    os.environ.setdefault("JINA_PORT", str(45678))
+
+    os.makedirs(os.environ["JINA_WORKSPACE"], exist_ok=True)
+    return
+
+
+def index_documents():
+    f = Flow().load_config(INDEX_FLOW_FILE_PATH)
+
+    with f:
+        f.index(input_fn, batch_size=8)
+
+
+def call_api(url, payload=None, headers=None):
+    if headers is None:
+        headers = {"Content-Type": "application/json; charset=utf-8"}
+    import requests
+
+    return requests.post(url, data=json.dumps(payload), headers=headers).json()
+
+
+def get_results(query, top_k=TOP_K):
+    return call_api(
+        "http://0.0.0.0:45678/api/search", payload={"top_k": top_k, "data": [query]}
+    )
+
+
+def get_flow():
+    f = Flow().load_config(QUERY_FLOW_FILE_PATH)
+    f.use_rest_gateway()
+    return f
+
+
+@pytest.fixture
+def queries_and_expected_replies():
+    return [
+        (
+            "Trudging slowly\n",
+            [
+                "Trudging slowly over wet sand. Back to the bench where your clothes were "
+                "stolen. This is a coastal town. That they forgot to close down. Armagedon - "
+                "come armagedon come armagedon come. Everyday is like sunday. Everyday is "
+                "silent and grey. Hide on a promanade. Etch on a post card:. How I dearly "
+                "wish I was not here. In the seaside town. That they forgot to bomb. Come, "
+                "come nuclear bomb!. Everyday is like sunday. Everyday is silent and grey. "
+                "Trudging back over pebbles and sand. And a strange dust lands on your hands. "
+                "(and on your face). Everyday is like sunday. Win yourself a cheap tray. "
+                "Share some grease tea with me. Everyday is silent and grey",
+                "These are. These are days you'll remember. Never before and never since, I "
+                "promise. Will the whole world be warm as this. And as you feel it,. You'll "
+                "know it's true. That you - you are blessed and lucky. It's true - that you. "
+                "Are touched by something. That will grow and bloom in you. These are days "
+                "you'll remember. When May is rushing over you. With desire to be part of the "
+                "miracles. You see in every hour. You'll know it's true. That you are blessed "
+                "and lucky. It's true that you are touched. By something that will grow and "
+                "bloom in you. These are days. These are the days you might fill. With "
+                "laughter until you break. These days you might feel. A shaft of light. Make "
+                "its way across your face. And when you do. You'll know how it was meant to "
+                "be. See the signs and know their meaning. You'll know how it was meant to "
+                "be. Hear the signs and know they're speaking. To you, to you",
+            ],
+        ),
```

Comment:
Maybe we can figure out a better way to include the expected results.

Reply:
One possibility would be: Just test the first 5 of each list. If they are the same, most probably everything is fine.

Reply:
I think we should index smaller lyrics. But also the tests shoould be run with `multi-chunk` queries to test the full power of the example and to have a good tests that heavily uses the `document recursivity`.

Reply:
at the end we should "mimic" the color behavior observed in the frontend

Reply:
I will dig in and implement the assertion on the chunk level